File: 221226/JavaScript/fruitSeller.js
export function sortInDescendingOrder(scores) {
  return scores.sort((a, b) => b - a);
}

export default function solution(k, numberInBox, scores) {
  if (numberInBox > scores.length) {
    return 0;
  }

  const descendingScores = sortInDescendingOrder(scores);

  return descendingScores.reduce((answer, score, index) => (
    index % numberInBox === numberInBox - 1
      ? answer + (score * numberInBox)
      : answer
  ), 0);
}

File: solutions/Previous/221226/Java/FruitSeller.java
import java.util.Arrays;

public class FruitSeller {
    public int solution(int k, int numberInBox, int[] scoresPrimitive) {
        if (numberInBox > scoresPrimitive.length) {
            return 0;
        }

        Integer[] scores = Arrays.stream(scoresPrimitive)
            .boxed()
            .toArray(Integer[]::new);

        sortInDescendingOrder(scores);

        int answer = 0;

        for (int i = numberInBox - 1; i < scores.length; i += numberInBox) {
            answer += scores[i] * numberInBox;
        }

        return answer;
    }

    public void sortInDescendingOrder(Integer[] scores) {
        Arrays.sort(scores, (a, b) -> b - a);
    }
}
